### PR TITLE
FW: Adding cmdlets for creating and deleting AD application & service principal

### PR DESCRIPTION
**Pull request by perseusCode**
This change contains following four cmdlets :-

a) New-AzureADApplication
b) Remove-AzureADApplication
c) New-AzureADServicePrincipal
d) Remove-AzureADServicePrincipal

These cmdlets are needed to maintain AD service princiapals. Tests for the same have also been added.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@perseusCode__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (mansikka). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by ogail** [comment]
add to whitelist

File: src/ResourceManager/Resources/Commands.Resources/ActiveDirectory/NewAzureADApplicationCommand.cs [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+namespace Microsoft.Azure.Commands.ActiveDirectory
+{
+    /// <summary>
+    /// Creates a new AD application.
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "AzureADApplication", DefaultParameterSetName = ParameterSet.ApplicationWithoutCredential), OutputType(typeof(PSADApplication))]
+    public class NewAzureADApplicationCommand : ActiveDirectoryBaseCmdlet
+    {
+        private string _KeyType = "AsymmetricX509Cert";
```
Comment (ogail) [inline src/ResourceManager/Resources/Commands.Resources/ActiveDirectory/NewAzureADApplicationCommand.cs]: nit: we don't use `_` convention in PowerShell

File: src/ResourceManager/Resources/Commands.Resources/ActiveDirectory/NewAzureADApplicationCommand.cs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        private string _KeyUsage = "Verify";
```
Comment (ogail) [inline src/ResourceManager/Resources/Commands.Resources/ActiveDirectory/NewAzureADApplicationCommand.cs]: You can remove these variables totally and set the properties to the default value in a constructor.

**Comment by ogail** [comment]
retest this please

**Comment by perseusCode** [comment]
Didn't get your comment, do you want me to run the unit tests again ?

**Comment by ogail** [comment]
@perseusCode that's a comment for the CI to kickoff the build again. Apparently it failed so please fix it.

**Comment by perseusCode** [comment]
The build break is because Microsoft.Graph.RBAC.dll 1.4 onwards are not compatible with Azure powershell. The SDK needs to be fixed to generate proper MAML library.

**Comment by ogail** [comment]
retest this please

**Comment by perseusCode** [comment]
Closing this as this PR is against master.

**Pull request closed without merging**